src/main/java/com/stuypulse/robot/subsystems/Climber.java

```diff
@@ -5,24 +5,155 @@
+ * @author Ca7Ac1(Ayan Chowdhury)
+ * @author marcjiang7(Marc Jiang)
+ * @author ambers7(Amber Shen)
+ * @author Souloutz(Howard Kong)
+ * @author jiayuyan0501(Jiayu Yan)
+ * @author ijiang05(Ian Jiang)
+ * @author TraceyLin (Tracey Lin)
+ * @author annazheng14(Anna Zheng)
+ * @author lonelydot(Raymond Zhang)
+ * @author andylin2004 (Andy Lin)
  */
 public class Climber extends SubsystemBase {
 
+    private Solenoid solenoidLong;
+    private Solenoid solenoidShort;
+
+    private DigitalInput limitSwitch;
+
+    private Solenoid stopper;
+
+    private RelativeEncoder encoder;
+
+    private CANSparkMax climber;
+
     public Climber() {
-        // Initialize Motors, etc... here!
+        climber = new CANSparkMax(Constants.Ports.Climber.MOTOR);
+        climber.setInverted(Constants.ClimberSettings.MOTOR_INVERTED);
+  
+        encoder = climber.getEncoder();
+        encoder.setPositionConversionFactor(Constants.ClimberSettings.CIRCUMFERENCE * Constants.ClimberSettings.GEAR_RATIO);
+
+        solenoidLong = new Solenoid(PneumaticsModuleType.CTREPCM, Constants.Ports.Climber.SOLENOID_LONG);
+        solenoidShort = new Solenoid(PneumaticsModuleType.CTREPCM, Constants.Ports.Climber.SOLENOID_SHORT);
+        stopper = new Solenoid(PneumaticsModuleType.CTREPCM, Constants.Ports.Climber.STOPPER);
+
+        limitSwitch = new DigitalInput(Constants.Ports.Climber.LIMIT_SWITCH);
+    }
+    
+    private void moveMotor(double speed) {
+        climber.set(speed);
+    }
+
+    public void liftUp() {
+        moveMotor(Constants.ClimberSettings.CLIMBER_DEFAULT_SPEED.get());
+    }
+
+    public void liftDown() {
+        moveMotor(-Constants.ClimberSettings.CLIMBER_DEFAULT_SPEED.get());
+    }
+
+    public void stop() {
+        moveMotor(0.0);
+    }
+    
+    public void liftUpSlow() {
+        moveMotor(Constants.ClimberSettings.CLIMBER_SLOW_SPEED.get());
+    }
+
+    public void liftDownSlow() {
+        moveMotor(-Constants.ClimberSettings.CLIMBER_SLOW_SPEED.get());
+    }
+
+    public void extendLong() {
+        solenoidLong.set(true);
+    }
+    
+    public void retractLong() {
+        solenoidLong.set(false);
+    }
+    
+    public void extendShort() {
+        solenoidShort.set(true);
+    }
+
+    private void retractShort() {
+        solenoidShort.set(false);
+    }
+
+    public void fullyRetract() {
+        retractShort();
+        retractLong();
+    }
+
+    public void fullyExtend() {
+        extendLong();
+        extendShort();
+    }
+
+    public boolean getLongExtended() {
+        return solenoidLong.get();
+    }
+
+    public boolean getShortExtended() {
+        return solenoidShort.get();
+    }
+
+    public double getEncoderDistance() {
+        return encoder.getPosition();
+    }
+
+    public void reset() {
+        encoder.setPosition(0.0);
+    }
+
+    public void getLimitSwitch() {
+        if (limitSwitch.get()) {
+            encoder.setPosition(0.0);
+            stop();
+        }
+    }
```

Comment:
i think the naming on this also needs to be improved. i imagine that you would want a separate method that simply returns if the limit switch has been pressed. 

the logic in here for homing the encoder (which is good) is for likely for periodic() instead of this method. also be careful with using stop (i actually don't know if it'll be bad because idk what commands will run at the same time and idk how the limit switch is placed / when it'll be in contact with the arm)